### ISSUE TITLE
Refactor

### ISSUE THREAD

**Issue by nshaud** [nshaud/DeepHyperX#35]
- [x] Rewrite disjoint sampling method
- [x] Move sklearn models into their own file
- [x] Rewrite the Dataset
- [x] Add parallelization option (see #32)
- [x] Use a unique IGNORED_INDEX value for all ignored pixels
- [x] Use `sklearn.metrics` everywhere needed (especially in validation)
- [x] Move data exploration/data visualization functions into their own file
- [x] Rewrite the `build_dataset` function
- [x] Main script uses a `main` function
- [x] Unify `val` and `test` function
- [ ] Deal with the varying tensor sizes when using : spectra (1D), images (2D), cubes (3D).
- [x] Unify segmentation and classification datasets
- [ ] Use `Sequential` API for simple models
- [x] Add --overlap options for training and test
- [ ] Use scheduler/auto LR reduction (see #22)
- [ ] Save output image after training
- [ ] Rewrite data augmentation as `torchvision.transforms` (see #33)
- [ ] Move downloaders into their own script
- [ ] Add other class balancing schemes (see #39)
- [ ] Add IoU/dice score loss
- [ ] Improve cross-validation support
- [ ] _Optional: Simplify dataset configuration_
- [ ] _Optional: Support defining a dataset as a collection of HSI images and GT masks_

**Comment by 3097108366** [comment]
Where is the output image after training saved,I really can't find it
